File: apps/desktop/src-tauri/src/command/feed_entry.rs
use colette_api::{
    feed_entry::{FeedEntry, FeedEntryListQuery, FeedEntryUpdate},
    Paginated, Session,
};
use colette_core::feed_entry::FeedEntryService;
use tauri::State;
use uuid::Uuid;

#[tauri::command]
pub async fn list_feed_entries(
    service: State<'_, FeedEntryService>,
    session: Session,
    query: FeedEntryListQuery,
) -> Result<Paginated<FeedEntry>, String> {
    let entries = service
        .list_feed_entries(query.into(), session.profile_id)
        .await
        .map_err(|e| e.to_string())?;

    Ok(entries.into())
}

#[tauri::command]
pub async fn get_feed_entry(
    service: State<'_, FeedEntryService>,
    session: State<'_, Session>,
    id: Uuid,
) -> Result<FeedEntry, String> {
    let entry = service
        .get_feed_entry(id, session.profile_id)
        .await
        .map_err(|e| e.to_string())?;

    Ok(entry.into())
}

#[tauri::command]
pub async fn update_feed_entry(
    service: State<'_, FeedEntryService>,
    session: State<'_, Session>,
    id: Uuid,
    data: FeedEntryUpdate,
) -> Result<FeedEntry, String> {
    let entry = service
        .update_feed_entry(id, data.into(), session.profile_id)
        .await
        .map_err(|e| e.to_string())?;

    Ok(entry.into())
}

File: apps/desktop/src/commands/feed-entry.ts
import {
  FeedEntry,
  type FeedEntryAPI,
  FeedEntryList,
  FeedEntryListQuery,
  FeedEntryUpdate,
} from '@colette/core'
import { invoke } from '@tauri-apps/api/core'

export class FeedEntryCommands implements FeedEntryAPI {
  list(query: FeedEntryListQuery): Promise<FeedEntryList> {
    return invoke('list_feed_entries', {
      query: FeedEntryListQuery.parse(query),
    }).then(FeedEntryList.parse)
  }

  get(id: string): Promise<FeedEntry> {
    return invoke('get_feed_entry', { id }).then(FeedEntry.parse)
  }

  update(id: string, data: FeedEntryUpdate): Promise<FeedEntry> {
    return invoke('update_feed_entry', {
      id,
      data: FeedEntryUpdate.parse(data),
    }).then(FeedEntry.parse)
  }
}
